atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableCreator.java

```diff
@@ -0,0 +1,137 @@
+    }
+
+    void createTables(Map<TableReference, byte[]> tableRefToMetadata) {
+        try {
+            clientPool.runWithRetry(client -> {
+                for (Map.Entry<TableReference, byte[]> entry : tableRefToMetadata.entrySet()) {
+                    createTable(entry.getKey(), entry.getValue(), client);
+                }
+                CassandraKeyValueServices.waitForSchemaVersions(config, client, "after adding the column family for "
+                        + "tables " + tableRefToMetadata.keySet() + " in a call to create tables");
+                return null;
+            });
+        } catch (Exception e) {
+            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+        }
+    }
+
+    private void createTable(TableReference tableRef, byte[] metadata, CassandraClient client) throws TException {
+        CqlQuery query = constructQuery(tableRef, metadata);
+        client.execute_cql3_query(query, Compression.NONE, org.apache.cassandra.thrift.ConsistencyLevel.QUORUM);
+    }
+
+    private CqlQuery constructQuery(TableReference tableRef, byte[] metadata) {
+        StringBuilder queryBuilder = new StringBuilder();
+
+        TableMetadata tableMetadata = CassandraKeyValueServices.getMetadataOrDefaultToGeneric(metadata);
+        boolean appendHeavyReadLight = tableMetadata.isAppendHeavyAndReadLight();
+
+        Map<String, String> settings = new HashMap<>();
+        settings.put("bloom_filter_fp_chance", falsePositive(tableMetadata.hasNegativeLookups(), appendHeavyReadLight));
+        settings.put("caching", "'KEYS_ONLY'");
+        settings.put("compaction", getCompaction(appendHeavyReadLight));
+        settings.put("compression", getCompression(tableMetadata.getExplicitCompressionBlockSizeKB()));
+        settings.put("gc_grace_seconds", Integer.toString(config.gcGraceSeconds()));
+        settings.put("id", "'%s'");
+        settings.put("populate_io_cache_on_flush", cachePriorityIsHottest(tableMetadata));
+        settings.put("speculative_retry", "'NONE'");
+
+        queryBuilder.append("CREATE TABLE IF NOT EXISTS \"%s\".\"%s\""
+                + " ( " + ROW + " blob, " + COLUMN + " blob, " + TIMESTAMP + " bigint, " + VALUE + " blob, "
+                + "PRIMARY KEY (" + ROW + ", " + COLUMN + ", " + TIMESTAMP + ")) "
+                + "WITH COMPACT STORAGE ");
+
+        settings.forEach((option, value) -> queryBuilder.append("AND " + option + " = " + value + " "));
+
+        queryBuilder.append("AND CLUSTERING ORDER BY (" + COLUMN + " ASC, " + TIMESTAMP + " ASC) ");
+
+        return CqlQuery.builder()
+                .safeQueryFormat(queryBuilder.toString())
+                .addArgs(
+                        SafeArg.of("keyspace", config.getKeyspaceOrThrow()),
+                        SafeArg.of("internalTableName", CassandraKeyValueServiceImpl.internalTableName(tableRef)),
+                        SafeArg.of("cfId", getUuidForTable(tableRef)))
+                .build();
+    }
```

Comment:
I defer to Cassops - but base Atlas explicitly sets fields to default values in `ColumnFamilyDefinitions` which we aren't setting here, like `dclocal_read_repair_chance`, `min_index_interval` etc. Is it safe now to not explicitly set these?